test/extended/templates/templateservicebroker_e2e.go

```diff
@@ -168,9 +177,18 @@ var _ = g.Describe("[templates] templateservicebroker end-to-end test", func() {
 			},
 		}))
 
-		o.Expect(templateInstance.Status.Conditions).To(o.HaveLen(1))
-		o.Expect(templateInstance.Status.Conditions[0].Type).To(o.Equal(templateapi.TemplateInstanceReady))
-		o.Expect(templateInstance.Status.Conditions[0].Status).To(o.Equal(kapi.ConditionTrue))
+		o.Expect(templateInstance.Status.Conditions).To(o.HaveLen(2))
+		o.Expect(templateInstance.HasCondition(templateapi.TemplateInstanceReady, kapi.ConditionTrue)).To(o.Equal(true))
+		o.Expect(templateInstance.HasCondition(templateapi.TemplateInstanceWaiting, kapi.ConditionFalse)).To(o.Equal(true))
```

Comment:
needs tests to cover:

1) seeing the template in a waiting state
2) case where the template fails.